crates/jstz_api/src/console.rs

```diff
@@ -23,97 +21,66 @@ use boa_engine::{
     Context, JsArgs, JsNativeError, JsResult, JsValue, NativeFunction,
 };
 use boa_gc::{empty_trace, Finalize, GcRefMut, Trace};
-use jstz_core::{host::HostRuntime, runtime, value::IntoJs};
-use jstz_crypto::{hash::Blake2b, public_key_hash::PublicKeyHash};
-use serde::{Deserialize, Serialize};
-
-use tezos_smart_rollup::prelude::debug_msg;
-
-pub const LOG_PREFIX: &str = "[JSTZ:SMART_FUNCTION:LOG] ";
-
-#[derive(Serialize, Deserialize)]
-pub struct LogRecord {
-    pub contract_address: PublicKeyHash,
-    pub request_id: String,
-    pub level: String,
-    pub text: String,
-}
-
-impl Display for LogRecord {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str(&serde_json::to_string(self).map_err(|_| std::fmt::Error)?)
-    }
-}
-
-impl FromStr for LogRecord {
-    type Err = serde_json::Error;
-    fn from_str(json: &str) -> Result<Self, Self::Err> {
-        serde_json::from_str(json)
+use jstz_core::value::IntoJs;
+use log::js_log;
+pub use log::{set_console_logger, ConsoleLog, LogData, LogLevel};
```

Comment:
Personal preference, but I usually prefer to read the module linearly, meaning that these statements would make more sense _after_ `mod log`